### ISSUE TITLE
Adding an opening string when connection

### ISSUE THREAD

**Issue by elico** [kklis/proxy#6]
Hey,

I wrote a golang proxy which sends a "first line" when initiating a connection to the remote server.
I have seen your proxy and I wanted to verify two things:
- Can your work be compiled for windows as executable?
- Can you add an option to add a "first line" that will be sent before splicing\forwarding between the two connections?

If you would be able to help me this it will help me to minimize the size of the binary from 2-4 MB to couple bytes.
The code I wrote is at:
https://gist.github.com/elico/dd09874b011f809e0ca74f89403c505c#file-vnc-id-proxy-go-L23

I will be thankful for any help.

**Comment by kklis** [comment]
It should compile for Windows on Cygwin with no problem, as described in the [Readme](https://github.com/kklis/proxy/blob/master/README.md).
I consider my work on proxy done for the moment. However, the code is open source and the license is quite liberal, so you are welcome to fork the repository and modify the code according to your needs.

**Comment by elico** [comment]
OK.
For now I managed to build my proxy in a smaller size so it's enough for now.
I will try to see if there is a tcp proxy code for windows which I can use but I doubt.

**Comment by ticpu** [comment]
@elico take a look at the latest commit on my fork at https://github.com/ticpu/proxy

This searches for a specific string in the packet and redirects the connection somewhere else if it matches. You can take this as an example for your own fork.

I used this proxy since it is one of the simplest implementation, however, on Windows, Cygwin will make the whole package bigger and using fork on Windows is sub-optimal.

**Comment by elico** [comment]
@ticpu I was hoping to have a simple proxy would be windows compatible but eventually it's not.
In any case I found a nice way to reduce the size of the binary I have compared to others on the wild.
I want it to be used for tightvnc and found that Teamviewer and other products which are similar are spreading a 10MB binary package so I released a 2MB one which works pretty nice:
- 800 KB tightvnc server
- 100 Bytes windows batch file
-  1 MB Golang Proxy which can eventually have access to FS and other sides of the system such as starting up a program or rebooting the machine.
- The rest is 7z SFX content So I am happy with what I have for now with hope it will work for quite some time.
